flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java

```diff
@@ -25,20 +30,39 @@
 
 public class FlutterDescriptionProvider implements ModuleDescriptionProvider {
 
-  public static List<FlutterGalleryEntry> getGalleryList() {
+  @Override
+  public Collection<? extends ModuleGalleryEntry> getDescriptions() {
+    return getGalleryList(false);
+  }
+
+  public static List<FlutterGalleryEntry> getGalleryList(boolean isNewProject) {
+    boolean projectHasFlutter = isNewProject;
     OptionalValueProperty<FlutterProjectModel> sharedModel = new OptionalValueProperty<>();
     ArrayList<FlutterGalleryEntry> res = new ArrayList<>();
-    res.add(new FlutterApplicationGalleryEntry(sharedModel));
-    res.add(new FlutterPluginGalleryEntry(sharedModel));
-    res.add(new FlutterPackageGalleryEntry(sharedModel));
+    if (!isNewProject) {
+      IdeFrame frame = IdeFocusManager.getGlobalInstance().getLastFocusedFrame();
+      Project project = frame == null ? null : frame.getProject();
+      if (project == null) return res;
+      for (Module module : FlutterModuleUtils.getModules(project)) {
+        if (FlutterModuleUtils.isFlutterModule(module)) {
+          projectHasFlutter = true;
+          break;
+        }
+      }
+    }
+    if (projectHasFlutter) {
+      // Makes no sense to add Flutter templates to Android projects...
+      res.add(new FlutterApplicationGalleryEntry(sharedModel));
+      res.add(new FlutterPluginGalleryEntry(sharedModel));
+      res.add(new FlutterPackageGalleryEntry(sharedModel));
+    }
+    if (System.getProperty("flutter.experimental.modules", null) != null) {
```

Comment:
Let's definitely plan on moving this option into user preferences (in future PRs).